api/v1alpha4/tags.go

```diff
@@ -136,6 +136,12 @@ const (
 	// See https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/
 	// for annotation formatting rules.
 	VMTagsLastAppliedAnnotation = "sigs.k8s.io/cluster-api-provider-azure-last-applied-tags-vm"
+
+	// RGTagsLastAppliedAnnotation is the key for the Azure Cluster object annotation
+	// which tracks the AdditionalTags for Resource Group which is part in the Azure Cluster.
```

Comment:
nit
```suggestion
	// which tracks the capz specific tags for Resource Group which includes the AdditionalTags 
	// set in AzureCluster, and AzureManagedControlPlane.
```

Reply:
I have to think about this change. This might be true with the current implementation, which isn't reflecting some of the changes that was discussed in this PR as I'm yet to work on updating the PR. Recently we had a conversation where we were talking about how the tags will be managed, specifically talking about how we will track tag deletion using the last-applied-tags annotation and what it means. I'll think on the implementation next week and get back. You can follow the conversation here - https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1721#issuecomment-934598442 and here - https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1721#issuecomment-934301110 . We settled on creating CAPZ default tags as part of resource creation in the appropriate resource service and then manage tag create, update and delete in tags service, and at first thought I think tags service will find created, update, deleted and finally update the last-applied-tags annotation with full desired spec - which currently seems to be default + additional tags, but I might change the implementation where the tag spec will just contain additional tags, then annotation would just track additional tags, then the above change won't be needed. But yeah, I'll first implement everything and then get back to this anyway, when the code is more finalized